### DISCUSSION TITLE
Cannot import 'from dagster import solid'

### DISCUSSION THREAD

**Discussion by JunaidRana** [dagster-io/dagster#21579] [Q&A]
I have just started using Dagster for one of our major projects, so I am setting up assets, pipelines, job,s etc. I am using docker to run this, and installing dagster inside my docker, I can import all these packages below in my code but the solid is not working for me.

Need help to resolve this.

Here are the import , which are working fine.

```
from dagster import job, Definitions , DependencyDefinition , repository , schedule
from dagster import AssetExecutionContext
from dagster.core.definitions import InputDefinition
from dagster import asset , job
from dagster import MaterializeResult, MetadataValue, asset , AssetExecutionContext
```

Getting following error for the solid import

`dagster._core.errors.DagsterImportError: Encountered ImportError: `cannot import name 'solid' from 'dagster' (/usr/local/lib/python3.10/site-packages/dagster/__init__.py)` while importing module main from file /opt/dagster/app/main.py. Local modules were resolved using the working directory `/opt/dagster/app`. If another working directory should be used, please explicitly specify the appropriate path using the `-d` or `--working-directory` for CLI based targets or the `working_directory` configuration option for `python_file`-based workspace targets.`

I am using 2 docker file so here is my docker file to install dagster (**Dockerfile_user_code**)

FROM python:3.10-slim

COPY requirements.txt /opt/dagster/app/
WORKDIR /opt/dagster/app

RUN apt-get update && apt-get install -y sqlite3 libsqlite3-dev

RUN pip install \
    dagster \
    dagster-postgres \
    dagster-docker \
    dagster-airbyte

Second docker file installing Dagster (**Dockerfile_dagster**)
FROM python:3.10-slim

RUN pip install \
    dagster \
    dagster-graphql \
    dagster-webserver \
    dagster-postgres \
    dagster-docker

ENV DAGSTER_HOME=/opt/dagster/dagster_home/

RUN mkdir -p $DAGSTER_HOME

COPY docker/dagster.yaml docker/workspace.yaml $DAGSTER_HOME

WORKDIR $DAGSTER_HOME

**Comment by cmpadden** [comment]
Hi @JunaidRana - solid has been deprecated for a long time, so it's odd for that to be referenced. I presume you have followed this guide on deploying to Docker?

https://docs.dagster.io/deployment/guides/docker#deploying-dagster-to-docker

I confirmed that the imports you've shared work as expected. However, I noticed your Dockerfiles are not appropriately formatted -- this could very well just be how they've rendered on GitHub. For example, for your _pip installs_, I would expect a `\` character:

```
RUN pip install \
    dagster \
    dagster-graphql \
    dagster-webserver \
    dagster-postgres \
    dagster-docker
```

Could you please confirm these files, and possibly cross-reference to the deployment docs?
